### PR TITLE
Exclude current article

### PR DESCRIPTION
**Pull request by pe7er**
This PR adds in the **Latest News Module** (mod_articles_news) an option **exclude the current article** to filter the current article from displaying in the module.

### Summary of Changes
If you configure the Latest News Module to display Articles from Category "A",
and the Module is displayed on an Article page (that is also in Category "A"),
the Article is displayed on the page and in the Module.

- This  PR adds an option in the Latest News Module to exclude the current article.
- The option is OFF by default (so the current behavior before installing the patch is default)
- It also checks for com_content being active to prevent that other components that use an "id" might interfere with the Latest News Module output.

### Testing Instructions

- Create  a Category "A"
- Add 5 different articles with different titles
- Create a menu item for Category blog and another for Single Article
- Assign the "Latest News" Module to the two pages, configure display title and change the limit to 5.
- If you are on an Article page, the com_content component article view will display the article, but it is also visible in the "Latest News" Module output.
- In the parameters of "Latest News" Module, switch on "Exclude current Article" and check the page again.

mod_articles_news-exclude-article

### CODE REVIEW

File: modules/mod_articles_news/helper.php [hunk 1]
```diff
@@ -61,6 +61,13 @@ public static function getList(&$params)
 		// Filer by tag
```
Comment (Quy) [inline modules/mod_articles_news/helper.php:62]:
```suggestion
		// Filter by tag
```

File: modules/mod_articles_news/mod_articles_news.xml [hunk 2]
```diff
@@ -186,6 +186,19 @@
...
 
+				<field
+						name="exclude_current"
+						type="radio"
+						label="MOD_ARTICLES_NEWS_FIELD_EXCLUDE_CURRENT_LABEL"
+						description="MOD_ARTICLES_NEWS_FIELD_EXCLUDE_CURRENT_DESC"
+						class="btn-group btn-group-yesno"
+						default="0"
+						filter="integer"
+				>
```
Comment (Quy) [inline modules/mod_articles_news/mod_articles_news.xml:197]:
```suggestion
					>
```

File: modules/mod_articles_news/mod_articles_news.xml [hunk 3]
```diff
@@ -186,6 +186,19 @@
 					<option value="1">MOD_ARTICLES_NEWS_VALUE_ONLY_SHOW_FEATURED</option>
 				</field>
 
+				<field
+						name="exclude_current"
```
Comment (Quy) [inline modules/mod_articles_news/mod_articles_news.xml:190]: Reduce lines 190-196 by one tab.

File: language/en-GB/en-GB.mod_articles_news.ini [hunk 4]
```diff
@@ -7,6 +7,8 @@ MOD_ARTICLES_NEWS="Articles - Newsflash"
 MOD_ARTICLES_NEWS_FIELD_FEATURED_DESC="Show or hide articles marked as featured."
 MOD_ARTICLES_NEWS_FIELD_FEATURED_LABEL="Featured Articles"
 MOD_ARTICLES_NEWS_FIELD_CATEGORY_DESC="Select Articles from a specific Category or a set of Categories. If no selection will show all categories as default."
+MOD_ARTICLES_NEWS_FIELD_EXCLUDE_CURRENT_LABEL="Exclude current Article"
```
Comment (Quy) [inline language/en-GB/en-GB.mod_articles_news.ini:10]:
```suggestion
MOD_ARTICLES_NEWS_FIELD_EXCLUDE_CURRENT_LABEL="Exclude Current Article"
```

**Comment by tecpromotion** [comment]
I have tested this item :white_check_mark: successfully on 1da3533d19edbb6e6383ca355053ddb9b87e05c1<br /><br />I have tested this successfully on Joomla! 3.9.2.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23575">issues.joomla.org/tracker/joomla-cms/23575</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 1da3533d19edbb6e6383ca355053ddb9b87e05c1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23575">issues.joomla.org/tracker/joomla-cms/23575</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23575">issues.joomla.org/tracker/joomla-cms/23575</a>.</sub>

File: language/en-GB/en-GB.mod_articles_news.ini (hunk 4 above)
Comment (HLeithner) [reply language/en-GB/en-GB.mod_articles_news.ini:10]: @brianteeman lower case is correct or?

File: modules/mod_articles_news/mod_articles_news.xml (hunk 3 above)
Comment (HLeithner) [reply modules/mod_articles_news/mod_articles_news.xml:190]: @pe7er can you please fix cs

**Review by HLeithner — CHANGES_REQUESTED** [review]

**Comment by brianteeman** [comment]
this should not be rtc

**Comment by pe7er** [comment]
Thank you all for testing!
This PR was for Joomla 3.x but won't be added in that series anymore.
I've redone this PR for Joomla 4: https://github.com/joomla/joomla-cms/pull/24672
Closing this PR...

**Pull request closed without merging**